src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -764,6 +772,11 @@ def process_removal_queue(self) -> None:
                     raise orchestrator.OrchestratorError(f"Could not purge {osd}")
                 logger.info(f"Successfully purged {osd} on {osd.hostname}")
 
+            if osd.zap:
+                # throws an exception if the zap fails
+                self.mgr.zap_osd(osd.osd_id)
```

Comment:
```suggestion
                self.zap_osd(osd.osd_id)
```
Why put this method into the module?